autorest/sender.go

```diff
@@ -168,7 +138,37 @@ func DoErrorUnlessStatusCode(codes ...int) SendDecorator {
 	}
 }
 
-// DoRetryForAttempts returns a SendDecorator that retries the request for up to the specified
+// DoPollForStatusCodes returns a SendDecorator that polls if the http.Response contains one of the
+// passed status codes. It expects the http.Response to contain a Location header providing the
+// URL at which to poll (using GET) and will poll until the time passed is equal to or greater than
+// the supplied duration. It will delay between requests for the duration specified in the
+// RetryAfter header or, if the header is absent, the passed delay. Polling may be canceled by
+// closing the optional channel on the http.Request.
+func DoPollForStatusCodes(duration time.Duration, delay time.Duration, codes ...int) SendDecorator {
+	if len(codes) == 0 {
```

Comment:
this if check and assignment seemed a bit unnecessary. nil slice and empty slice often behaves the same.